Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -4170,7 +4170,6 @@ static void adjustSettingsForLockdownMode(Settings& settings, const WebPreferenc
 #if ENABLE(WEB_AUDIO)
     settings.setWebAudioEnabled(false);
 #endif
-    settings.setDownloadableBinaryFontsEnabled(false);
```

Comment:
Instead of deleting `settings.setDownloadableBinaryFontsEnabled`, change to:

`settings.setShouldAllowLimitedUserInstalledFonts(true);`

Reply:
Should it be a setting (Settings.yaml) instead of a WebPreference (WebPreferences.yaml) ?

Reply:
Oh! Yes, a Setting would be the right thing.

Reply:
Should we simply replace one flag by the other and always allow to download but if the new flag is on we make the "allowed fonts check" , or you meant having 2 flags to describe different situations (DownloadableBinaryFontsEnabled, ShouldAllowLimitedUserInstalledFonts):
1. (false, false) : will not even try to download.
2. (false, true)  : will download, but just load the allowed fonts.
3. (true, false)  : will download and allow any fonts (skip the check for allowed fonts).
4. (true, true)   : should never happen, but if it does, the same as [3].

Reply:
Maybe we should make it a tri-state value:

DownloadableBinaryFontAllowedTypes:
  type: uint32_t
  webcoreOnChange: setNeedsRecalcStyleInAllFrames
  defaultValue:
    WebKitLegacy:
      default: 2
    WebKit:
      default: 2
    WebCore:
      default: 2

Then you could make a "DownloadableBinaryFontAllowedTypes enum:

enum class DownloadableBinaryFontAllowedTypes : uint8_t {
    None = 0,
    Restricted = 1 << 0,
    Any = 1 << 1
};

'Any' would work like the current `setDownloadableBinaryFontsEnabled(true)`, 'None' would work like `setDownloadableBinaryFontsEnabled(false)`.

I don't think this is API exposed to clients, so I doubt there would be much compat risk.